internal/lookout/repository/query.go

```diff
@@ -50,3 +57,317 @@ func (r *SQLJobRepository) GetQueueStats() ([]*lookout.QueueInfo, error) {
 	}
 	return result, nil
 }
+
+func (r *SQLJobRepository) GetJobsInQueue(opts *lookout.GetJobsInQueueRequest) ([]*lookout.JobInfo, error) {
+	rows, err := r.queryJobsInQueue(opts)
+	if err != nil {
+		return nil, err
+	}
+
+	result := parseJobsInQueueRows(rows)
+	return result, nil
+}
+
+type jobsInQueueRow struct {
+	JobId     string
+	Queue     string
+	Owner     string
+	JobSet    string
+	Priority  sql.NullFloat64
+	Submitted pq.NullTime
+	Cancelled pq.NullTime
+	JobJson   sql.NullString
+	RunId     sql.NullString
+	Cluster   sql.NullString
+	Node      sql.NullString
+	Created   pq.NullTime
+	Started   pq.NullTime
+	Finished  pq.NullTime
+	Succeeded sql.NullBool
+	Error     sql.NullString
+}
+
+func (r *SQLJobRepository) queryJobsInQueue(opts *lookout.GetJobsInQueueRequest) ([]*jobsInQueueRow, error) {
+	query := makeGetJobsInQueueQuery(opts)
+	rows, err := r.db.Query(query)
+	if err != nil {
+		return nil, err
+	}
+
+	joinedRows := make([]*jobsInQueueRow, 0)
+	for rows.Next() {
+		row := &jobsInQueueRow{}
+		err := rows.Scan(
+			&row.JobId,
+			&row.Owner,
+			&row.JobSet,
+			&row.Priority,
+			&row.Submitted,
+			&row.Cancelled,
+			&row.JobJson,
+			&row.RunId,
+			&row.Cluster,
+			&row.Node,
+			&row.Created,
+			&row.Started,
+			&row.Finished,
+			&row.Succeeded,
+			&row.Error,
+		)
+		if err != nil {
+			return nil, err
```

Comment:
Idle question, is it better to return what you have gathered and the error, or nil and the error?